controlplane/kubeadm/internal/webhooks/kubeadm_control_plane_test.go

```diff
@@ -732,6 +761,25 @@ func TestKubeadmControlPlaneValidateUpdate(t *testing.T) {
 	unsetTimeouts.Spec.KubeadmConfigSpec.InitConfiguration.Timeouts = nil
 	unsetTimeouts.Spec.KubeadmConfigSpec.JoinConfiguration.Timeouts = nil
 
+	validUpdateCertificateValidityPeriod := before.DeepCopy()
+	validUpdateCertificateValidityPeriod.Spec.KubeadmConfigSpec.ClusterConfiguration.CertificateValidityPeriodDays = 150
+
+	invalidUpdateCACertificateValidityPeriodDays := before.DeepCopy()
+	invalidUpdateCACertificateValidityPeriodDays.Spec.KubeadmConfigSpec.ClusterConfiguration = &bootstrapv1.ClusterConfiguration{
+		CACertificateValidityPeriodDays: 730,
+	}
+
+	invalidRolloutBeforeCertificatesExpiryDays := before.DeepCopy()
+	invalidRolloutBeforeCertificatesExpiryDays.Spec.Rollout.Before.CertificatesExpiryDays = 8
+	invalidRolloutBeforeCertificatesExpiryDays.Spec.KubeadmConfigSpec.ClusterConfiguration.CertificateValidityPeriodDays = 7
+
+	invalidCertificateValidityPeriodDays := before.DeepCopy()
+	invalidCertificateValidityPeriodDays.Spec.KubeadmConfigSpec.ClusterConfiguration.CertificateValidityPeriodDays = 400
+
+	invalidCertificateValidityPeriod := before.DeepCopy()
+	invalidCertificateValidityPeriod.Spec.KubeadmConfigSpec.ClusterConfiguration.CACertificateValidityPeriodDays = 0
+	invalidCertificateValidityPeriod.Spec.KubeadmConfigSpec.ClusterConfiguration.CertificateValidityPeriodDays = 3651
```

Comment:
```suggestion
```
Unused

Reply:
apologies, Missed it in a hurry, Updated it. Thanks